libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -557,6 +557,82 @@ using __type_front = __type_at_c<0, _List>;
 template <class _List>
 using __type_back = __type_at_c<_List::__size - 1, _List>;
 
+//! \brief A pair of types
+template <class _First, class _Second>
+struct _CCCL_TYPE_VISIBILITY_DEFAULT __type_pair
+{
+  using __first _CCCL_NODEBUG_ALIAS  = _First;
+  using __second _CCCL_NODEBUG_ALIAS = _Second;
+};
+
+//! \brief Retreive the first of a pair of types
+//! \pre \c _Pair is a specialization of \c __type_pair
+template <class _Pair>
+using __type_pair_first = typename _Pair::__first;
+
+//! \brief Retreive the second of a pair of types
+//! \pre \c _Pair is a specialization of \c __type_pair
+template <class _Pair>
+using __type_pair_second = typename _Pair::__second;
```

Comment:
Nitpick: Should this be
```suggestion
using __type_pair_second _CCCL_NODEBUG_ALIAS = typename _Pair::__second;
```